fhttp/httprunner.go

```diff
@@ -61,8 +61,9 @@ func (httpstate *HTTPRunnerResults) Run(t int) {
 // options.
 type HTTPRunnerOptions struct {
 	periodic.RunnerOptions
-	HTTPOptions        // Need to call Init() to initialize
-	Profiler    string // file to save profiles to. defaults to no profiling
+	HTTPOptions               // Need to call Init() to initialize
+	Profiler           string // file to save profiles to. defaults to no profiling
+	AllowInitialErrors bool   // whether initial errors don't cause an abort
```

Comment:
curious: what do initial errors signify?

Reply:
like a 503 during the first fetch, the previous (and now still default cmd line) behavior is to abort on non 200

the UI will now not abort on error - that way you can use say `&status=503:40` to get 40% 503s and not risk having it stop right away if one of the initial requests hit that probability

Do you think there is a way I can clarify the description/comment ?

Reply:
This more-or-less aligns with my understanding. It seems like if any of the request number of threads fails an initial request, fortio will fail altogether. Is that a correct regurgitation?

Do we want to default to allowing connection errors?  Is the concern repeatedly failing connections (and a need to fail-fast)?

In other words, should this be `FailFast` instead of `AllowErrors` ?  I don't know that it really matters, just curious if you ever intend to run it with this option turned off.

Reply:
I guess the reason for the failfast current/past behavior is to quickly  and effectively catch a typo in a url (404) or an error in setup quickly (service doesn't exist)

if you start a 1h run to have it scroll with errors for 1h it's usually not what you want

with istio though it seems failfast means it'd give up too early :-) so maybe I should change the default

Reply:
ha. no that's ok. if running for an hour, we would want to catch a silly error first. 👍